js_tests/admin/SelectFilter2.test.js

```diff
@@ -16,3 +16,131 @@ QUnit.test('init', function(assert) {
     assert.equal($('.selector-remove').text(), "Remove");
     assert.equal($('.selector-clearall').text(), "Remove all");
 });
+
+QUnit.test('filtering available options', function(assert) {
+    const $ = django.jQuery;
+    $('<form><select multiple id="select"></select></form>').appendTo('#qunit-fixture');
+    $('<option value="foo" title="Foo">Foo</option>').appendTo('#select');
+    $('<option value="bar" title="Bar">Bar</option>').appendTo('#select');
+    $('<option value="baz" title="Baz">Baz</option>').appendTo('#select');
+    SelectFilter.init('select', 'items', 0);
+    assert.equal($('#select_from option').length, 3);
+    assert.equal($('#select_to option').length, 0);
+    const done = assert.async();
+    const event = new KeyboardEvent('keyup', {'key': 'b'});
+    $('#select_input').val('b');
+    SelectFilter.filter_key_up(event, 'select', '_from');
+    setTimeout(() => {
+        assert.equal($('#select_from option').length, 2);
+        assert.equal($('#select_to option').length, 0);
+        assert.equal($('#select_from option')[0].value, 'bar');
+        assert.equal($('#select_from option')[1].value, 'baz');
+        done();
+    });
+});
+
+QUnit.test('filtering selected options', function(assert) {
+    const $ = django.jQuery;
+    $('<form><select multiple id="select"></select></form>').appendTo('#qunit-fixture');
+    $('<option selected value="foo" title="Foo">Foo</option>').appendTo('#select');
+    $('<option selected value="bar" title="Bar">Bar</option>').appendTo('#select');
+    $('<option selected value="baz" title="Baz">Baz</option>').appendTo('#select');
+    SelectFilter.init('select', 'items', 0);
+    assert.equal($('#select_from option').length, 0);
+    assert.equal($('#select_to option').length, 3);
+    const done = assert.async();
+    const event = new KeyboardEvent('keyup', {'key': 'b'});
+    $('#select_selected_input').val('b');
+    SelectFilter.filter_key_up(event, 'select', '_to', '_selected_input');
+    setTimeout(() => {
+        assert.equal($('#select_from option').length, 0);
+        assert.equal($('#select_to option').length, 2);
+        assert.equal($('#select_to option')[0].value, 'bar');
+        assert.equal($('#select_to option')[1].value, 'baz');
+        done();
+    });
+});
+
+QUnit.test('filtering available options to nothing', function(assert) {
```

Comment:
`'filtering available options to nothing'` works without the patch and should be committed separately.